File: Frontend/angular-app/src/app/modules/accounting/entities/AutoVoucherFormula.ts
export class AutoVoucherFormula {
  public id!:number
  public voucherTypeId!:number
  public sourceVoucherTypeId!:number
  public voucherTypeTitle!:string
  public sourceVoucherTypeTitle!:string
  public orderIndex!:number
  public debitCreditStatus!:number
  public accountHeadId!:number
  public rowDescription!:string
  public formula!:string
  public conditions!:string
  public parsedFormula!:VoucherFormula[]
}


export class VoucherFormula {
  public property!:string
  public value!:Value
}

export class Value {
  public text!:string
  public properties!:Property[]
}

export class Property {
  public name!:string
  public aggregateFunction!:string
}

export class Condition {
  public expression!:string
  public properties!:Property[]
}

File: Backend/Microservices/Accounting/Eefa.Accounting.Data/Entities/AutoVoucherFormula.cs
using Library.Common;

namespace Eefa.Accounting.Data.Entities
{

    public partial class AutoVoucherFormula : BaseEntity
    {
        public int VoucherTypeId { get; set; } = default!;
        public int SourceVoucherTypeId { get; set; } = default!;

        /// <summary>
        /// ترتیب آرتیکل سند حسابداری
        /// </summary>
        public int OrderIndex { get; set; } = default!;

        /// <summary>
        /// وضعیت مانده حساب
        /// </summary>
        public byte DebitCreditStatus { get; set; } = default!;

        /// <summary>
        /// کد سطح
        /// </summary>
        public int AccountHeadId { get; set; } = default!;

        /// <summary>
        /// توضیحات سطر
        /// </summary>
        public string? RowDescription { get; set; }

        /// <summary>
        /// ستونهای مقصد
        /// </summary>
        public string? Formula { get; set; }

        /// <summary>
        /// شرط
        /// </summary>
        public string? Conditions { get; set; }
        public string? GroupBy { get; set; }

        public virtual AccountHead AccountHead { get; set; } = default!;
        public virtual User CreatedBy { get; set; } = default!;
        public virtual User? ModifiedBy { get; set; } = default!;
        public virtual Role OwnerRole { get; set; } = default!;
        public virtual CodeVoucherGroup VoucherType { get; set; } = default!;
        public virtual CodeVoucherGroup SourceVoucherType { get; set; } = default!;

    }
}
